File: codegen/codegen-erlang/src/test/resources/ir/handler_discovery_resolve_impl.expected.erl
resolve_impl(Impl) ->
    case code:ensure_loaded(Impl) of
        {module, Impl} ->
            Callbacks = basic_service_behaviour:behaviour_info(callbacks),
            Handlers = maps:from_list([{Fun, make_handler(Impl, Fun)} || {Fun, 3} <- Callbacks, erlang:function_exported(Impl, Fun, 3)]),
            {ok, Handlers};
        {error, _} ->
            {error, {impl_not_loaded, Impl}}
    end.

File: codegen/codegen-elixir/src/test/resources/ir/handler_discovery_resolve_impl.expected.ex
defp resolve_impl(impl) do
  case Code.ensure_loaded(impl) do
    {:module, _} ->
      handlers =
        for {fun, 3} <-
          BasicServiceBehaviour.callbacks(),
          function_exported?(impl, fun, 3),
          into: %{} do
          {fun, Function.capture(impl, fun, 3)}
        end
      {:ok, handlers}
    {:error, _} -> {:error, {:impl_not_loaded, impl}}
  end
end
